src/scripts/tag-image.sh

```diff
@@ -1,10 +1,12 @@
 #!/bin/bash
 
-SOURCE_TAG=$(eval echo "$ORB_EVAL_SOURCE_TAG")
-TARGET_TAG=$(eval echo "$ORB_EVAL_TARGET_TAG")
+ORB_VAL_REGISTRY_URL="$(circleci env subst "$ORB_VAL_REGISTRY_URL")"
+ORB_VAL_IMAGE="$(circleci env subst "$ORB_VAL_IMAGE")"
+
+SOURCE_TAG="$(circleci env subst "$ORB_EVAL_SOURCE_TAG")"
+TARGET_TAG="$(circleci env subst "$ORB_EVAL_TARGET_TAG")"
```

Comment:
While this is the recommended way of doing parameter expansion, it is causing the tests to break due to a substring operation in the [tag parameter](https://github.com/CircleCI-Public/gcp-gcr-orb/blob/master/.circleci/test-deploy.yml#L27):

```yaml
tag: ${CIRCLE_SHA1:0:7}.$CIRCLE_BUILD_NUMBER
```

`eval echo` can handle that without a problem, but this is not something `envsubst` is capable of or should be capable of handling. Hence the error.

I gravitate towards leaving `eval echo` for now since removing it will result in a loss of functionality, therefore, a breaking change.

We plan on rewriting the GCP orbs in the near future. When we do so, we will likely move forward with `envsubst` as part of the several breaking changes.

Once you roll this back, the PR will be good to go!